hash_reader.go

```diff
@@ -230,7 +228,13 @@ func (hashReader *hashReader) traverse() error {
 // The following implements hashValue interface.
 
 func (hashReader *hashReader) getFieldName() string {
-	return hashReader.FieldName()
+	fieldName := hashReader.FieldName()
+
+	if fieldName == nil {
```

Comment:
We should not return an empty string here.
Nil means fieldname does not exist, whereas "" is a valid fieldname

Reply:
Addressed.